core/domain/user_jobs_continuous_test.py

```diff
@@ -591,3 +597,458 @@ def test_basic_computation_works_if_collection_is_deleted(self):
+            self.MULTIPLIER)
+
+        # Verify that the impact score matches the expected.
+        self._run_manager()
+        user_stats_model = user_models.UserStatsModel.get(self.user_a_id)
+        self.assertEqual(user_stats_model.impact_score, expected_user_impact_score)
+
+    def test_standard_user_impact_calculation_multiple_explorations(self):
+        """Test that a user who is a contributor on two explorations that:
+        - have a number of ratings for that exploration above the treshold
+        for the scaler for number of ratings
+        - have an average rating above the minimum average rating
+        - have a number of playthroughs > 1
+        is assigned the correct impact score by the
+        UserImpactMRJobManager.
+        """
+        # Sign up a user and have them create two explorations.
+        self.user_a_id = self._sign_up_user(
+            self.USER_A_EMAIL, self.USER_A_USERNAME)
+        self.exploration_1 = self._create_exploration(
+            self.EXP_ID_1, self.user_a_id)
+        self.exploration_2 = self._create_exploration(
+            self.EXP_ID_2, self.user_a_id)
+        # Give these explorations as many ratings as necessary to avoid
+        # the scaler for number of ratings.
+        self._rate_exploration(
+            self.exploration_1.id, self.NUM_RATINGS_SCALER_CUTOFF, self.ABOVE_MIN_RATING)
+        self._rate_exploration(
+            self.exploration_2.id, self.NUM_RATINGS_SCALER_CUTOFF, self.ABOVE_MIN_RATING)
+        # Give these explorations more than one playthrough (so
+        # ln(num_completions != 0).
+        self._complete_exploration(self.exploration_1, self.MIN_NUM_COMPLETIONS)
+        self._complete_exploration(self.exploration_2, self.MIN_NUM_COMPLETIONS)
+
+        # The user impact score should be the rounded sum of these two impacts
+        # (2 * the same impact).
+        expected_user_impact_score = round(2 *
+            math.log(self.MIN_NUM_COMPLETIONS) *
+            (self.ABOVE_MIN_RATING - self.MIN_AVERAGE_RATING) *
+            self.MULTIPLIER)
+
+        # Verify that the impact score matches the expected.
+        self._run_manager()
+        user_stats_model = user_models.UserStatsModel.get(self.user_a_id)
+        self.assertEqual(user_stats_model.impact_score, expected_user_impact_score)
+
+    def test_only_yield_when_impact_greater_than_zero(self):
+        """Tests that map only yields an impact score for an
+        exploration when the impact score is greater than 0."""
+        # Sign up a user and have them create an exploration.
+        self.user_a_id = self._sign_up_user(
+            self.USER_A_EMAIL, self.USER_A_USERNAME)
+        self.exploration = self._create_exploration(
+            self.EXP_ID_1, self.user_a_id)
+
+        # Use mock impact scores to verify that map only yields when
+        # the impact score > 0.
+        with self.swap(self.impact_mr_job_manager,
```

Comment:
Do something like this:

    # The following lines check that the results list contains exactly one value.
    next(results)
    with self.assertsRaises(StopIteration):
        next(results)